### DISCUSSION TITLE
Netty TCP Client Unable to identify that TCP server has closed the connection

### DISCUSSION THREAD

**Discussion by nsarathk** [netty/netty#14212] [Q&A]
am using Netty 4.1.104.Final version in Java 17 and spring boot 2.7.18, The TCP Client i have is unable to read / know the tcp server has closed the connection for me to initiate a new connection, for now i am using the idle handler to find the idle timeout to create a new connection everytime the application finds no messages from the server i am closing and opening the connection

**Comment by violetagg** [comment]
you can use exactly idle timeout or TCP keep-alive

**Reply by nsarathk** [reply]
I am able to idle it out but keep alive of tcp server is not recognised by netty tcp client is there a way to recognise or read it at tcp client end ?
